src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -1931,6 +1939,74 @@ public List<String> adminApprovalFields() {
 		}
 		return new ArrayList<>();
 	}
+
+
+	/**
+	 * Updates the user profile with version 2.
+	 *
+	 * @param request   The request containing profile update data.
+	 * @param userToken The user token for authentication.
+	 * @param authToken The authentication token.
+	 * @param rootOrgId The root organization ID.
+	 * @return SBApiResponse object containing the response of the profile update.
+	 * @throws Exception if any error occurs during the profile update process.
+	 */
+	@Override
+	public SBApiResponse profileUpdateV2(Map<String, Object> request, String userToken, String authToken, String rootOrgId) throws Exception {
+		// Extracting the request map from the input request
+		Map<String, Object> requestMap = (Map<String, Object>) request.get(Constants.REQUEST);
+		// Validating the token from the request map
+		if (validateToken((String) requestMap.get("token"))) {
+			// If the token is valid, delegate the profile update to the profileUpdate method
+			return profileUpdate(request, userToken, authToken, rootOrgId);
+		}
+		// If token validation fails, return null
+		return null;
+	}
+
+
+	/**
+	 * Validates the given JWT token.
+	 *
+	 * @param token The JWT token to validate.
+	 * @return True if the token is valid; false otherwise.
+	 * @throws ProfileRequestException if any error occurs during token validation.
+	 */
+	public boolean validateToken(String token) throws ProfileRequestException {
+		try {
+			// Parsing the token and extracting its subject (claims)
+			String tokens = Jwts.parser().setSigningKey(serverConfig.getSecretKeyPrimaryEmailValidation()).parseClaimsJws(token).getBody().getSubject();
+			// Parsing the subject (claims) into a map using Jackson ObjectMapper
+			Map<String, String> parsedMap = mapper.readValue(tokens, new TypeReference<Map<String, String>>() {
+			});
+			// Checking if the 'contextType' key exists and is not empty
+			return parsedMap.get(Constants.CONTEXT_TYPE) != null && !parsedMap.get(Constants.CONTEXT_TYPE).isEmpty();
+		} catch (ExpiredJwtException e) {
+			// Handling ExpiredJwtException
+			throw new ProfileRequestException("ExpiredJwt exception has occurred: " + e.getMessage());
```

Comment:
throw new Exception(errMsg, e) -- should be fine for all of this exceptions

Reply:
1. It would throw Exception with the error message.